internal/providers/providers.go

```diff
@@ -9,12 +9,37 @@ import (
 
 	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw"
 	"github.com/databricks/terraform-provider-databricks/internal/providers/sdkv2"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
 	"github.com/hashicorp/terraform-plugin-framework/providerserver"
 	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
 	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
 	"github.com/hashicorp/terraform-plugin-mux/tf6muxserver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+type serverOptions struct {
+	sdkV2Provider           *schema.Provider
+	pluginFrameworkProvider provider.Provider
+}
+
+type ServerOption interface {
+	Apply(*serverOptions)
+}
+
+type sdkV2ProviderOption struct {
+	sdkV2Provider *schema.Provider
+}
+
+func (o *sdkV2ProviderOption) Apply(options *serverOptions) {
+	options.sdkV2Provider = o.sdkV2Provider
+}
+
+// WithSdkV2Provider takes in a *schema.Provider from sdk v2 and returns a ServerOption.
+// This is typically used in acceptance test for a test step to have a custom provider override.
+func WithSdkV2Provider(sdkV2Provider *schema.Provider) ServerOption {
```

Comment:
This seems to be only used in internal/acceptance/mws_workspaces_test.go? If yes, then probably best to move it to providers_test_utils.go

Reply:
I think this needs to stay here since it is part of the (internal) public interface for GetProviderServer. Alternatively, we could have two copies of `GetProviderServer()`, one with only the context argument and a customizable one, but that seems redundant to me.